File: minimal/typescript/src/tools/bash.ts
import type { CoreTool } from "../core/types.js";

export const bashTool: CoreTool = {
  name: "bash",
  description: "Execute a shell command in the workspace.",
  inputSchema: {
    type: "object",
    properties: {
      command: { type: "string", description: "Shell command to run." },
    },
    required: ["command"],
  },
};

File: minimal/rust/src/tools/bash.rs
use crate::types::Tool;
use serde_json::json;

pub fn bash_tool() -> Tool {
    Tool {
        name: "bash".to_string(),
        description: "Execute a shell command in the workspace.".to_string(),
        input_schema: json!({
            "type": "object",
            "properties": {
                "command": {
                    "type": "string",
                    "description": "Shell command to run."
                }
            },
            "required": ["command"]
        }),
    }
}
